File: evcs-station/src/main/java/com/evcs/station/entity/Station.java
package com.evcs.station.entity;

import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.IdType;
import lombok.Data;
import java.time.LocalDateTime;

/**
 * 充电站实体
 */
@Data
@TableName("charging_station")
public class Station {
    @TableId(value = "station_id", type = IdType.AUTO)
    private Long stationId;
    private Long tenantId;
    private String stationCode;
    private String stationName;
    private String address;
    private Double latitude;
    private Double longitude;
    private Integer status; // 1-启用 0-停用
    private String province;
    private String city;
    private String district;
    
    // 统计字段 - 通过 JOIN 查询计算，不存储在表中
    @TableField(exist = false)
    private Integer totalChargers;
    @TableField(exist = false)
    private Integer availableChargers;
    @TableField(exist = false)
    private Integer chargingChargers;
    @TableField(exist = false)
    private Integer faultChargers;
    
    private Long createBy;
    private Long updateBy;
    private LocalDateTime createTime;
    private LocalDateTime updateTime;
    private Integer deleted;
}

File: evcs-admin/src/api/station.ts
import request from '../utils/request'

/**
 * 充电站相关接口
 */

// 充电站列表查询参数
export interface StationQueryParams {
  stationName?: string
  province?: string
  city?: string
  status?: number
  current?: number
  size?: number
}

// 充电站信息
export interface Station {
  id: number
  stationCode: string
  stationName: string
  province: string
  city: string
  district?: string
  address: string
  longitude?: number
  latitude?: number
  contactName?: string
  contactPhone?: string
  totalChargers: number
  availableChargers: number
  status: number
  tenantId: number
  createTime?: string
  updateTime?: string
}

// 充电站表单数据
export interface StationForm {
  stationCode: string
  stationName: string
  province: string
  city: string
  district?: string
  address: string
  longitude?: number
  latitude?: number
  contactName?: string
  contactPhone?: string
  status: number
}

/**
 * 获取充电站列表
 */
export function getStationList(params: StationQueryParams) {
  return request({
    url: '/station/list',
    method: 'get',
    params
  })
}

/**
 * 获取充电站详情
 */
export function getStationDetail(id: number) {
  return request({
    url: `/station/${id}`,
    method: 'get'
  })
}

/**
 * 新增充电站
 */
export function createStation(data: StationForm) {
  return request({
    url: '/station',
    method: 'post',
    data
  })
}

/**
 * 更新充电站
 */
export function updateStation(id: number, data: StationForm) {
  return request({
    url: `/station/${id}`,
    method: 'put',
    data
  })
}

/**
 * 删除充电站
 */
export function deleteStation(id: number) {
  return request({
    url: `/station/${id}`,
    method: 'delete'
  })
}

/**
 * 获取充电站下的充电桩列表
 */
export function getStationChargers(stationId: number) {
  return request({
    url: `/station/${stationId}/chargers`,
    method: 'get'
  })
}
